renderdoc/driver/vulkan/vk_initstate.cpp

```diff
@@ -104,6 +106,13 @@ bool WrappedVulkan::Prepare_InitialState(WrappedVkRes *res)
       return Prepare_SparseInitialState((WrappedVkImage *)res);
     }
 
+#if ENABLED(RDOC_NEW_IMAGE_STATE)
+    LockedImageStateRef state = FindImageState(im->id);
+
+    // if the image has no memory bound, nothing is to be fetched
+    if(!state || !state->isMemoryBound)
```

Comment:
I added a check here--if the image has any foreign/external owned subresources, then no initial contents are saved.

Ideally/eventually, we probably want to save initial contents for the subresources that are not owned by foreign queues (and maybe also only save initial contents for those subresources that actually need it, once we go down that road)

Reply:
Yeh in theory we could save any non-foreign subresources to be slightly more correct but I suspect (without proving this definitively) that almost all uses of foreign/external queues are either for resources with only one subresource, or they pass the whole resource. It may not be worth the risk of obscure bugs, I'd probably at least wait until someone has a non-theoretical need for that.